parsons/databases/db_sync.py

```diff
@@ -56,9 +58,13 @@ def table_sync_full(self, source_table, destination_table, if_exists='drop',
         if destination_tbl.exists:
             if if_exists == 'drop':
                 destination_tbl.drop()
-            elif if_exists == 'truncate':
-                self._check_column_match(source_tbl, destination_tbl)
-                destination_tbl.truncate()
+            elif if_exists in ('truncate','drop_if_needed'):
+                try:
+                    self._check_column_match(source_tbl, destination_tbl)
+                    destination_tbl.truncate()
+                except:
```

Comment:
Doesn't this except clause make `truncate` and `drop_if_needed` exactly the same? i.e. I'd only do this for `drop_if_needed`, and allow `truncate` to fail if truncation doesn't work.

Reply:
You are right! Fixing!